lua/telescope/config.lua

```diff
@@ -49,6 +49,10 @@ config.descriptions = {}
 function config.set_defaults(defaults)
   defaults = defaults or {}
 
+  if defaults.layout_default then
+    error("'layout_default' is no longer a valid key. See 'layout_config'")
```

Comment:
`See 'layout_config'` is not a super helpful error message IMO.

I was thinking as well, perhaps we should just warn them for now and then do `defaults.layout_config = defaults.layout_defaults` or similar.

This way we don't crash everyone's telescope right away on download. (we can error in one week or something)